### PR TITLE
fix(radio-group): better alignment for horizontal radios

### LINKED ISSUE #2364: [Radio] UI distorts when align="horizontal"
## Current Behavior
UI distorts when align="horizontal" 

Steps to Reproduce: 

- Open - [https://baseweb.design/components/radio/](https://baseweb.design/components/radio/)
- Select `align="horizontal"`

image

## Expected Behavior
UI should render properly

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.21.1  |
| browser |     Chrome    |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

### PR DESCRIPTION
**Pull request by eekbatani**
Fixes #2364 

#### Description

The marginBottom of a radio button component was being set to 'null' in the case that the button had a description with it. This behavior is only correct for vertical alignment, and thus resulting in the distorted output when horizontally aligning the buttons, due to the absence of the bottom margin. 

I added a condition to the check that checks if the alignment is not set to horizontal. This way, if the alignment of the radio buttons are horizontal, the bottom margin will not be removed and the buttons will be correctly aligned. 
I added Another conditional property for the horizontal alignment, and adds a left margin if it is true. This is to ensure the radio buttons are not stuck together as they were previously. The following screenshot demonstrates the corrected output. 

image

#### Scope

- [x] Patch: Bug Fix
- [ ] Minor: New Feature
- [ ] Major: Breaking Change

### CODE REVIEW

**Comment by gergelyke** [comment]
thank you @eekbatani

File: src/radio/styled-components.js [hunk 1]
```diff
@@ -137,7 +137,8 @@ export const Root = styled<StylePropsT>('label', props => {
     alignItems: 'center',
     cursor: $disabled ? 'not-allowed' : 'pointer',
     marginTop: sizing.scale200,
-    marginBottom: $hasDescription ? null : sizing.scale200,
+    marginLeft: ($align !== 'horizontal') ? sizing.scale200: null,
```
Comment (chasestarr) [inline src/radio/styled-components.js]: Thanks for the contribution. When inspecting the screenshot changes [here](https://github.com/eekbatani/baseweb/pull/1/files?short_path=19699b9#diff-19699b9cc01a7b617c8f43c7dd260e60) There should not be margin-left applied when the alignment is vertical. Should the conditional on this line be inverted?

Comment (nadiia) [reply src/radio/styled-components.js]: Also, the first radio still shouldn't have left margin in horizontal alignment. Could you apply `marginRight` instead? Could also be `:first-child` rule but that would rely on no additional wrapper markup.

Comment (eekbatani) [reply src/radio/styled-components.js]: You're right, I'll make the changes.

**Comment by gergelyke** [comment]
@eekbatani would you mind fixing the eslint errors? you can reproduce them locally by running `yarn test`

**Comment by eekbatani** [comment]
> @eekbatani would you mind fixing the eslint errors? you can reproduce them locally by running `yarn test`


> @eekbatani would you mind fixing the eslint errors? you can reproduce them locally by running `yarn test`

Sure, I didn't see any on my end initially but I can see the issue now. I'll go ahead and fix them.

**Comment by gergelyke** [comment]
@eekbatani it seems like eslint is failing - can you look into that please?

**Comment by chasestarr** [comment]
> @eekbatani it seems like eslint is failing - can you look into that please?

below are the lint errors:
```
/baseui/src/radio/styled-components.js
--
  | 140:18  error  Replace `($align·===·'horizontal')·?·sizing.scale200` with `$align·===·'horizontal'·?·sizing.scale200·`              prettier/prettier
  | 141:18  error  Replace `·($hasDescription)·&&·($align·!==·'horizontal')·` with `⏎······$hasDescription·&&·$align·!==·'horizontal'`  prettier/prettier
```

**Comment by eekbatani** [comment]
> $hasDescription·&&·$align·!==·'horizontal'
Sure, I'll make the changes. Also, I did install the prettier extension but I'm something must have gone wrong on my end since it didn't pick up the issue.

File: src/radio/styled-components.js [hunk 2]
```diff
@@ -137,7 +137,8 @@ export const Root = styled<StylePropsT>('label', props => {
     alignItems: 'center',
     cursor: $disabled ? 'not-allowed' : 'pointer',
     marginTop: sizing.scale200,
-    marginBottom: $hasDescription ? null : sizing.scale200,
+    marginRight: $align === 'horizontal' ? sizing.scale200 : null,
```
Comment (nadiia) [inline src/radio/styled-components.js]: Would you mind adding rtl support for it?

**Comment by nadiia** [comment]
Horizontal positioning in the yard looks missaligned.
Screen Shot 2019-12-05 at 11 23 35 AM

**Comment by gergelyke** [comment]
@eekbatani are you planning to working on this pr? happy to take it over if you don't have the time

**Comment by eekbatani** [comment]
> @eekbatani are you planning to working on this pr? happy to take it over if you don't have the time

Yeah sure. I'm just in my final exam week for school so I'm low on time this week. You can go ahead and take over if you'd like to get it finish it off sooner.

**Comment by gergelyke** [comment]
Closing in favor of https://github.com/uber/baseweb/pull/2660

**Pull request closed without merging**